docs/docs/troubleshooting-common-errors.md

```diff
@@ -0,0 +1,219 @@
+
+Some ideas for debugging these errors is to verify the following:
+
+- you are using a transformer plugin (like `gatsby-transformer-yaml`) the data you need is pulled in using a source plugin (like `gatsby-source-filesystem`)
+
+```javascript:title=gatsby-config.js
+{
+  plugins: [
+    `gatsby-transformer-yaml`,
+    {
+      resolve: `gatsby-source-filesystem`,
+      options: {
+        path: `./src/data/`, // location of yaml files
+      },
+    },
+  ]
+}
+```
+
+- the structure of your content being sourced matches your GraphQL schema as well as the way you are querying the data
+
+Comparing your GraphQL query to your site's schema in [GraphiQL at `localhost:8000/___graphql`](http://localhost:8000/___graphql) and whatever plugin or code you are using to source data is a great way to find these errors as they should all express the data in the same shape.
+
+- a source plugin you are using, or your own implementation of the [`sourceNodes` API](/docs/node-apis/#sourceNodes) isn't misconfigured
+
+## Errors using gatsby-image and sharp
+
+Gatsby's image processing is broken up into different packages which need to work together to source images and transform them into different optimized versions. You might run into these errors getting them to play together nicely.
+
+### Field "image" must not have a selection since type "String" has no subfields
+
+This errror message `Field "image" must not have a selection since type "String" has no subfields.` comes up when a GraphQL query is trying to query a field for subfields, but none exist. This generally happens when plugins that are used together are added in the `gatsby-config` in the wrong order, or haven't been added at all.
+
+The query is trying to access fields that don't exist because they weren't set up at build time. In the following code, a query is looking to find the subfield `childImageSharp` of the `image` field, like the error states. The problematic GraphQL schema looks like this:
+
+```graphql
+allMdx {
+  nodes {
+    id
+    title
+    image
+  }
+}
+```
+
+And the expected GraphQL schema looks this:
+
+```graphql
+allMdx {
+  nodes {
+    id
+    title
+    image {
+      childImageSharp {
+        fluid {
+          srcSet
+        }
+      }
+    }
+  }
+}
+```
+
+In the first code example, the `image` field was not modified (using Gatsby terminology, transformed) by a plugin to add subfields, so it would only return a string. `gatsby-plugin-sharp` and `gatsby-transformer-sharp` can be included before other plugins that would manipulate or create image nodes (like `gatsby-source-filesystem` or `gatsby-source-contentful`) to ensure that they are present before Gatsby tries to modify them and add the needed fields like `childImageSharp`.
```

Comment:
```suggestion
In the first code example, the `image` field was not transformed (_modified_) by a plugin to add subfields, so it would only return a string. `gatsby-plugin-sharp` and `gatsby-transformer-sharp` can be included before other plugins that would manipulate or create image nodes (like `gatsby-source-filesystem` or `gatsby-source-contentful`) to ensure that they are present before Gatsby tries to modify them and add the needed fields like `childImageSharp`.
```